api/src/conf.rs

```diff
@@ -1,4 +1,100 @@
 use config::{Config, File};
+use dotenvy;
+use regex::Regex;
+use std::collections::HashMap;
+use std::env;
+use tracing::{debug, info, warn};
+
+#[derive(thiserror::Error, Debug)]
+pub enum ExpandError {
+    #[error("Missing environment variable '{var_name}' for placeholder '{placeholder}'")]
+    MissingVariable {
+        var_name: String,
+        placeholder: String,
+    },
+}
+
+/// Loads environment variables from a .env file if it exists.
+/// This function should be called early in the application startup,
+/// typically before any configuration loading.
+///
+/// The function looks for .env files in this order:
+/// 1. .env in the current working directory
+/// 2. .env in the parent directory (useful when running from subdirs)
+/// 3. If none found, continues without error (env vars may be set by system)
+pub fn load_dotenv() {
+    // Try current directory first
```

Comment:
that does make me wonder how we should handle config locations. right now, the path to the config.yaml is often a parameter we pass to binaries we run. But I have a feeling that the convention might be different for .env files, that are just expected to be where you run your command from, correct?